R/test-threads.R

```diff
@@ -330,90 +330,59 @@ adjust_estimation_options <- function(.mods, .cap_iterations){
     map(.mods, function(.mod){
 
       mod_path <- get_model_path(.mod)
-      mod_lines <- mod_path %>% readLines() %>%
-        suppressSpecificWarning("incomplete final line found")
-
+      ctl <- nmrec::read_ctl(mod_path)
 
       # Identify EST Blocks
-      est_idxs <- get_est_idx(mod_lines)
-
-      est_block <- map(seq_along(est_idxs), ~{
-        mod_lines[est_idxs[[.x]]]
-      })
-
-      est_idxs <- unlist(est_idxs)
+      ests <- nmrec::select_records(ctl, "est")
 
-      if(is.null(est_idxs)){
+      if(rlang::is_empty(ests)){
         warning(glue("No Estimation line found in {basename(mod_path)}, so there are no options to cap"))
         return(NULL)
       }
 
-      for(i in seq_along(est_block)){
-        est_block.i <- est_block[[i]]
-
-        max_match <- "MAX(EVAL(S)?)?=\\d+" ## need to vet this regex
-        max_detect <- str_detect(est_block.i, max_match)
-        niter_detect <- str_detect(est_block.i, "NITER=\\d+")
-        nburn_detect <- str_detect(est_block.i, "NBURN=\\d+")
-        if (any(max_detect)) {
-          est_block.i[max_detect] <- replace_est_opt(est_block.i[max_detect], max_match, .cap_iterations)
-        } else if(any(niter_detect)) {
-          est_block.i[niter_detect] <- replace_est_opt(est_block.i[niter_detect], "NITER=\\d+", .cap_iterations)
-          if (any(nburn_detect) && !any(str_detect(est_block.i, "NBURN=0\\b"))) {
-            est_block.i[nburn_detect] <- replace_est_opt(est_block.i[nburn_detect], "NBURN=\\d+", .cap_iterations)
-          }else if(!any(nburn_detect)){
+      for(i in seq_along(ests)){
+        est_block.i <- ests[[i]]
+        est_block.i$parse()
+
+        max_detect <- any(purrr::map_lgl(est_block.i$get_options(), ~ .x$name == "maxevals"))
```

Comment:
This and the similar lines below are fine as is, but I'm guessing they came before the switch to making `get_record_option` return `NULL` in nmrec's [47228bd](https://github.com/metrumresearchgroup/nmrec/commit/47228bddb5736f371781c25bd48ac63daa6ca476). With that, this could also be written as is

```R
max_detect <- !is.null(get_record_option(est_block.i, "maxevals"))
```

But, again, I think either way is okay.